src/main/java/com/algorand/algosdk/logic/SourceMap.java

```diff
@@ -0,0 +1,112 @@
+package com.algorand.algosdk.logic;
+
+import java.lang.Integer;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * SourceMap class provides parser for source map from  
+ * algod compile endpoint 
+ */
+public class SourceMap {
+
+    public int version;
+    public String file;
+    public String[] sources;
+    public String[] names;
+    public String mappings;
+
+    public HashMap<Integer, Integer> pcToLine;
+    public HashMap<Integer, ArrayList<Integer>> lineToPc;
+
+    public SourceMap(HashMap<String,Object> sourceMap) {
+        int version = (int) sourceMap.get("version");
+        if(version != 3){
+            throw new IllegalArgumentException("Only source map version 3 is supported");
+        }
+        this.version = version;
+
+        this.file = (String) sourceMap.get("file");
+        this.mappings = (String) sourceMap.get("mappings");
+
+        this.lineToPc = new HashMap<>();
+        this.pcToLine = new HashMap<>();
+
+        Integer lastLine = 0;
+        String[] vlqs = this.mappings.split(";");
+        for(int i=0; i<vlqs.length; i++){
+            ArrayList<Integer> vals = VLQDecoder.decodeSourceMapLine(vlqs[i]);
+
+        	// If the vals length >= 3 the lineDelta
+            if(vals.size() >= 3){
+                lastLine = lastLine + vals.get(2);
+            }
+
+            if(!this.lineToPc.containsKey(lastLine)){
+                this.lineToPc.put(lastLine, new ArrayList<Integer>());
+            }
+
+            ArrayList<Integer> currList = this.lineToPc.get(lastLine);
+            currList.add(i);
+            this.lineToPc.put(lastLine, currList);
+
+            this.pcToLine.put(i, lastLine);
+        }
+
+    }
+
+    public Integer getLineForPc(Integer pc) {
+        if(!this.pcToLine.containsKey(pc)){
+            return 0;
+        }
```

Comment:
@barnjamin What do you think of the following suggestion?

Since we're on Java 7, https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html is unavailable and I don't feel creating a custom type is warranted.  Instead, I propose returning `null` rather than 0 when key does not exist. 

I think the proposed behavior is more inline with other SDKs and less prone to misinterpretation. 
 
```suggestion
```

Reply:
Optionally - If you take the _suggestion_, I'd consider adding a Javadoc comment highlighting the return value semantics.